e2e/src/usecases/SecureSend.js

```diff
@@ -16,77 +16,147 @@ const faker = require('@faker-js/faker')
+      await device.uninstallApp()
+      await device.installApp()
+      await launchApp({
+        newInstance: true,
+        permissions: { notifications: 'YES', contacts: 'YES' },
+        launchArgs: {
+          statsigGateOverrides: `use_new_send_flow=false,use_viem_for_send=true`,
+        },
       })
+      await quickOnboarding(SAMPLE_BACKUP_KEY_VERIFIED)
+    })
 
-      it('Send cUSD to phone number with multiple mappings', async () => {
-        let randomContent = faker.lorem.words()
-        await waitFor(element(by.id('HomeAction-Send')))
-          .toBeVisible()
-          .withTimeout(30000)
-        await element(by.id('HomeAction-Send')).tap()
-        await waitFor(element(by.id('SendSearchInput'))).toBeVisible()
-
-        await element(by.id('SearchInput')).tap()
-        await element(by.id('SearchInput')).replaceText(VERIFIED_PHONE_NUMBER)
-        await element(by.id('RecipientItem')).tap()
-
-        // Select the currency
-        await waitFor(element(by.id('cUSDTouchable'))).toBeVisible()
-        await element(by.id('cUSDTouchable')).tap()
-
-        // Enter the amount and review
-        await inputNumberKeypad(AMOUNT_TO_SEND)
-        await element(by.id('Review')).tap()
-
-        // Use the last digits of the account to confirm the sender.
-        await waitFor(element(by.id('confirmAccountButton'))).toBeVisible()
-        await element(by.id('confirmAccountButton')).tap()
-        // TODO: test case for AddressValidationType.PARTIAL but relies on mapping phone number to another address with unique last 4 digits
-        // for (let index = 0; index < 4; index++) {
-        //   const character = LAST_ACCOUNT_CHARACTERS[index]
-        //   await element(by.id(`SingleDigitInput/digit${index}`)).replaceText(character)
-        // }
-        await element(by.id('ValidateRecipientAccount/TextInput')).replaceText(
-          SAMPLE_WALLET_ADDRESS_VERIFIED_2
-        )
-
-        // Scroll to see submit button
-        await scrollIntoView('Submit', 'KeyboardAwareScrollView', 50)
-        await element(by.id('ConfirmAccountButton')).tap()
-
-        // Write a comment.
-        await addComment(randomContent)
-
-        // Confirm and input PIN if necessary.
-        await element(by.id('ConfirmButton')).tap()
-        await enterPinUiIfNecessary()
-
-        // Return to home screen.
-        await waitFor(element(by.id('HomeAction-Send')))
-          .toBeVisible()
-          .withTimeout(30 * 1000)
-
-        await waitFor(element(by.text(`${randomContent}`)))
-          .toBeVisible()
-          .withTimeout(60000)
+    it('Send cUSD to phone number with multiple mappings', async () => {
+      let randomContent = faker.lorem.words()
+      await waitFor(element(by.id('HomeAction-Send')))
+        .toBeVisible()
+        .withTimeout(30000)
+      await element(by.id('HomeAction-Send')).tap()
+      await waitFor(element(by.id('SendSearchInput'))).toBeVisible()
```

Comment:
These waitFor(s) need a `.withTimeout(<time in ms>)`. There is also a helper function `waitForElementId` in the `e2e/src/utils/utils.js` that could also be used instead.

Reply:
This waitFor needs a [`.withTimeout()`](https://github.com/nubank/detox/blob/master/docs/APIRef.waitFor.md#tobevisible).